TestFX not compatible with OpenJFX Monocle

Hi guys,

I wanted to use an upgraded version of Monocle because I have a lot of java.nio.BufferOverflowException

So I decided to import the latest one here:
```
<!-- https://mvnrepository.com/artifact/org.testfx/openjfx-monocle -->
<dependency>
    <groupId>org.testfx</groupId>
    <artifactId>openjfx-monocle</artifactId>
    <version>8u76-b04</version>
</dependency>
```

But then I have this exception : 
```
java.lang.RuntimeException: java.lang.RuntimeException: java.lang.IllegalAccessException: Class org.testfx.toolkit.impl.ApplicationLauncherImpl can not access a member of class com.sun.glass.ui.monocle.HeadlessPlatform with modifiers ""
	at org.testfx.util.WaitForAsyncUtils.waitFor(WaitForAsyncUtils.java:161)
	at org.testfx.api.FxToolkit.waitForLaunch(FxToolkit.java:322)
	at org.testfx.api.FxToolkit.registerPrimaryStage(FxToolkit.java:131)
	at com.nellarmonia.installer.MatTestsIT.setupClass(MatTestsIT.java:71)
Caused by: java.lang.RuntimeException: java.lang.IllegalAccessException: Class org.testfx.toolkit.impl.ApplicationLauncherImpl can not access a member of class com.sun.glass.ui.monocle.HeadlessPlatform with modifiers ""
	at org.testfx.toolkit.impl.ApplicationLauncherImpl.initMonocleHeadless(ApplicationLauncherImpl.java:80)
	at org.testfx.toolkit.impl.ApplicationLauncherImpl.launch(ApplicationLauncherImpl.java:56)
	at org.testfx.toolkit.impl.ToolkitServiceImpl.lambda$setupPrimaryStage$0(ToolkitServiceImpl.java:70)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at org.testfx.util.WaitForAsyncUtils.callCallableAndSetFuture(WaitForAsyncUtils.java:364)
	at org.testfx.util.WaitForAsyncUtils.lambda$async$1(WaitForAsyncUtils.java:88)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.IllegalAccessException: Class org.testfx.toolkit.impl.ApplicationLauncherImpl can not access a member of class com.sun.glass.ui.monocle.HeadlessPlatform with modifiers ""
	at sun.reflect.Reflection.ensureMemberAccess(Reflection.java:102)
	at java.lang.Class.newInstance(Class.java:436)
	at org.testfx.toolkit.impl.ApplicationLauncherImpl.assignHeadlessPlatform(ApplicationLauncherImpl.java:101)
	at org.testfx.toolkit.impl.ApplicationLauncherImpl.initMonocleHeadless(ApplicationLauncherImpl.java:74)
	at org.testfx.toolkit.impl.ApplicationLauncherImpl.launch(ApplicationLauncherImpl.java:56)
	at org.testfx.toolkit.impl.ToolkitServiceImpl.lambda$setupPrimaryStage$0(ToolkitServiceImpl.java:70)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at org.testfx.util.WaitForAsyncUtils.callCallableAndSetFuture(WaitForAsyncUtils.java:364)
	at org.testfx.util.WaitForAsyncUtils.lambda$async$1(WaitForAsyncUtils.java:88)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

Is it normal? I'm using TestFX 4.0.5-alpha and using Monocle 8u20-b26, also putting this when launching the tests :
```
System.setProperty("testfx.robot", "glass");
        System.setProperty("testfx.headless", "true");
        System.setProperty("prism.order", "sw");
        System.setProperty("prism.text", "t2k");
        System.setProperty("headless.geometry", "1600x1200-64");

```

Well now I have this error:
```
java.lang.RuntimeException: java.lang.RuntimeException: java.lang.AbstractMethodError: com.sun.glass.ui.monocle.NativePlatform.createInputDeviceRegistry()Lcom/sun/glass/ui/monocle/input/InputDeviceRegistry;
	at org.testfx.util.WaitForAsyncUtils.waitFor(WaitForAsyncUtils.java:284)
	at org.testfx.api.FxToolkit.waitForLaunch(FxToolkit.java:323)
	at org.testfx.api.FxToolkit.registerPrimaryStage(FxToolkit.java:132)
	at com.nellarmonia.installer.FiltersTestIT.setupClass(FiltersTestIT.java:73)
Caused by: java.lang.RuntimeException: java.lang.AbstractMethodError: com.sun.glass.ui.monocle.NativePlatform.createInputDeviceRegistry()Lcom/sun/glass/ui/monocle/input/InputDeviceRegistry;
	at com.sun.javafx.tk.quantum.QuantumToolkit.startup(QuantumToolkit.java:267)
	at com.sun.javafx.application.PlatformImpl.startup(PlatformImpl.java:211)
	at com.sun.javafx.application.LauncherImpl.startToolkit(LauncherImpl.java:675)
	at com.sun.javafx.application.LauncherImpl.launchApplication1(LauncherImpl.java:695)
	at com.sun.javafx.application.LauncherImpl.lambda$launchApplication$155(LauncherImpl.java:182)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.AbstractMethodError: com.sun.glass.ui.monocle.NativePlatform.createInputDeviceRegistry()Lcom/sun/glass/ui/monocle/input/InputDeviceRegistry;
	at com.sun.glass.ui.monocle.NativePlatform.getInputDeviceRegistry(NativePlatform.java:58)
	at com.sun.glass.ui.monocle.MonocleApplication.<init>(MonocleApplication.java:75)
	at com.sun.glass.ui.monocle.MonoclePlatformFactory.createApplication(MonoclePlatformFactory.java:43)
	at com.sun.glass.ui.Application.run(Application.java:146)
	at com.sun.javafx.tk.quantum.QuantumToolkit.startup(QuantumToolkit.java:257)
	at com.sun.javafx.application.PlatformImpl.startup(PlatformImpl.java:211)
	at com.sun.javafx.application.LauncherImpl.startToolkit(LauncherImpl.java:675)
	at com.sun.javafx.application.LauncherImpl.launchApplication1(LauncherImpl.java:695)
	at com.sun.javafx.application.LauncherImpl.lambda$launchApplication$155(LauncherImpl.java:182)
	at java.lang.Thread.run(Thread.java:745)

```

Using Monocle 8u76-b04 and your version of TestFX..

Allright I have made a very simple Maven Project with only one test here : https://github.com/Maxoudela/MonocleTest

When I run it with JDK 8u92, I have this exception :
```
-------------------------------------------------------
 T E S T S
-------------------------------------------------------
Running BaseTest
Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.167 sec <<< FAILURE! - in BaseTest
BaseTest  Time elapsed: 0.166 sec  <<< ERROR!
java.lang.RuntimeException: java.lang.IllegalStateException: Monocle headless platform not found
	at org.testfx.util.WaitForAsyncUtils.waitFor(WaitForAsyncUtils.java:161)
	at org.testfx.api.FxToolkit.waitForLaunch(FxToolkit.java:264)
	at org.testfx.api.FxToolkit.registerPrimaryStage(FxToolkit.java:126)
	at BaseTest.setupClass(BaseTest.java:35)
Caused by: java.lang.IllegalStateException: Monocle headless platform not found
	at org.testfx.toolkit.impl.ApplicationLauncherImpl.initMonocleHeadless(ApplicationLauncherImpl.java:75)
	at org.testfx.toolkit.impl.ApplicationLauncherImpl.launch(ApplicationLauncherImpl.java:54)
	at org.testfx.toolkit.impl.ToolkitServiceImpl.lambda$setupPrimaryStage$53(ToolkitServiceImpl.java:70)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at org.testfx.util.WaitForAsyncUtils.callCallableAndSetFuture(WaitForAsyncUtils.java:363)
	at org.testfx.util.WaitForAsyncUtils.lambda$async$8(WaitForAsyncUtils.java:88)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.ClassNotFoundException: com.sun.glass.ui.monocle.headless.HeadlessPlatform
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:264)
	at org.testfx.toolkit.impl.ApplicationLauncherImpl.assignHeadlessPlatform(ApplicationLauncherImpl.java:98)
	at org.testfx.toolkit.impl.ApplicationLauncherImpl.initMonocleHeadless(ApplicationLauncherImpl.java:72)
	at org.testfx.toolkit.impl.ApplicationLauncherImpl.launch(ApplicationLauncherImpl.java:54)
	at org.testfx.toolkit.impl.ToolkitServiceImpl.lambda$setupPrimaryStage$53(ToolkitServiceImpl.java:70)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at org.testfx.util.WaitForAsyncUtils.callCallableAndSetFuture(WaitForAsyncUtils.java:363)
	at org.testfx.util.WaitForAsyncUtils.lambda$async$8(WaitForAsyncUtils.java:88)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)


```

If you could fork it and build it and confirm you don't see any errors.

--- a/subprojects/testfx-core/src/main/java/org/testfx/toolkit/impl/ApplicationLauncherImpl.java
+++ b/subprojects/testfx-core/src/main/java/org/testfx/toolkit/impl/ApplicationLauncherImpl.java
@@ -16,6 +16,7 @@
  */
 package org.testfx.toolkit.impl;
 
+import java.lang.reflect.Constructor;
 import java.lang.reflect.Field;
 import java.util.Objects;
 import javafx.application.Application;
@@ -40,9 +41,9 @@ public class ApplicationLauncherImpl implements ApplicationLauncher {
 
     private static final String NATIVE_PLATFORM_FACTORY =
         "com.sun.glass.ui.monocle.NativePlatformFactory";
-    private static final String HEADLESS_NATIVE_PLATFORM =
+    private static final String HEADLESS_PLATFORM =
         "com.sun.glass.ui.monocle.headless.HeadlessPlatform";
-    private static final String HEADLESS_U40_NATIVE_PLATFORM =
+    private static final String HEADLESS_PLATFORM_U40 =
         "com.sun.glass.ui.monocle.HeadlessPlatform";
 
     //---------------------------------------------------------------------------------------------
@@ -98,12 +99,14 @@ public class ApplicationLauncherImpl implements ApplicationLauncher {
                                  throws Exception {
         Class<?> nativePlatformFactoryClass = Class.forName(NATIVE_PLATFORM_FACTORY);
         try {
-            Object nativePlatformImpl = Class.forName(HEADLESS_U40_NATIVE_PLATFORM).newInstance();
-            assignPrivateStaticField(nativePlatformFactoryClass, "platform", nativePlatformImpl);
+            Constructor<?> nativePlatformCtor = Class.forName(HEADLESS_PLATFORM_U40).getDeclaredConstructor();
+            nativePlatformCtor.setAccessible(true);
+            assignPrivateStaticField(nativePlatformFactoryClass, "platform", nativePlatformCtor.newInstance());
         }
         catch (ClassNotFoundException exception) {
-            Object nativePlatformImpl = Class.forName(HEADLESS_NATIVE_PLATFORM).newInstance();
-            assignPrivateStaticField(nativePlatformFactoryClass, "platform", nativePlatformImpl);
+            Constructor<?> nativePlatformCtor = Class.forName(HEADLESS_PLATFORM).getDeclaredConstructor();
+            nativePlatformCtor.setAccessible(true);
+            assignPrivateStaticField(nativePlatformFactoryClass, "platform", nativePlatformCtor.newInstance());
         }
     }
 

--- a/subprojects/testfx-core/testfx-core.gradle
+++ b/subprojects/testfx-core/testfx-core.gradle
@@ -26,5 +26,5 @@ dependencies {
     testCompile "org.mockito:mockito-core:2.3.5"
     testCompile "org.hamcrest:hamcrest-library:1.3"
     testCompile "org.controlsfx:controlsfx:8.40.12"
-    testCompile "org.testfx:openjfx-monocle:1.8.0_20"
+    testCompile "org.testfx:openjfx-monocle:8u76-b04"
 }

